pkg/proctree/export.go

```diff
@@ -0,0 +1,214 @@
+package proctree
+
+/*
+Package proctree is used to create and maintain a process tree using Tracee's events.
+
+The process tree exports to the user only read operations - the user can query for a processes and
+threads information.
+However, the process tree itself is rather complicated, and shouldn't be changed by unsupported
+methods.
+Currently, the only supported way to update and change the process tree is via events. The process
+tree is aware of Tracee's events, and know how to update itself using events. It can be done
+using a pipeline or directly using the appropriate method.
+
+# Synchronization
+
+The tree usage is not synchronized.
+Process information might be requested some time after it was updated, or even deleted.
+To address this issue, the tree:
+*	Makes processes available even after they are exited and should be deleted for some time.
+*	Exports processes and threads information according to a specific time, instead of giving the
+	entire information and leaving the user to process it.
+
+# Events feeding edge cases handling
+
+The overall design of the tree is taking into consideration that events are not guaranteed to be
+ordered, nor they are guaranteed to arrive at all.
+Moreover, the tree needs to take into account that it starts running in the middle of the system
+runtime. As a result, it missed some processes fork, exec and exit events.
+To withstand this issues, the tree do the following things:
+*	Limit the amount of nodes it keeps, to avoid leakage when missing exit events.
+*	Take information from all events, even those which are not related to process and thread life
+	cycle.
+*	Try cleaning processes with PID which is found to be reused by new fork events.
+*	Initializing parent processes nodes, even with almost no information, to establish connections
+	between known nodes.
+*	The tree is *not* trying to create one big tree, but instead has the forest architecture.
+	As it doesn't have all the history, it can't connect all nodes, and so it connects only nodes
+	it knows are connected.
+*/
+
+import (
+	gocontext "context"
+
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/pkg/utils/types"
+	"github.com/aquasecurity/tracee/types/datasource"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
```

Comment:
ProcessTree struct should be here, together with `NewProcessTree()` as they are exported. Also, it might be better to have the type declaration closer to the documentation above.

Reply:
Also, I'm not sure I like the tree.go + export.go. I would keep all in the same file (its easier for readers to understand what is going on, as you already have the ProcessTree methods also split into `events_processing.go`, `exec_processing.go` and ` exit_processing.go` (those do make sense to have).